carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -1388,19 +1440,18 @@ namespace carma_wm
           auto curr_hour_boost = curr_day_boost + boost::posix_time::hours(hours_of_day);
 
           min_end_time += lanelet::time::durationFromSec(lanelet::time::toSec(curr_hour_boost));
-          ROS_DEBUG_STREAM("New min_end_time: " << std::to_string(lanelet::time::toSec(min_end_time)));
         }
 
         auto last_time_difference = sim_.last_seen_state_[curr_intersection.id.id][current_movement_state.signal_group].first - min_end_time;  
-        bool is_duplicate = last_time_difference.total_milliseconds() >= -30 && last_time_difference.total_milliseconds() <= 30;
+        bool is_duplicate = last_time_difference.total_milliseconds() >= -500 && last_time_difference.total_milliseconds() <= 500;
```

Comment:
Can I get a brief explanation of this change?

Reply:
The SPAT message's timestamp actually drifts every time with 0.1s. Having 0.03s check before was not anywhere near. So I had made it 0.5s to make it even more robust thinking even yellow light has to be at least 2.0s